src/Servers/Kestrel/Core/src/KestrelServer.cs

```diff
@@ -182,8 +167,10 @@ public async Task StopAsync(CancellationToken cancellationToken)
                 var tasks = new Task[_transports.Count];
                 for (int i = 0; i < _transports.Count; i++)
                 {
-                    tasks[i] = _transports[i].UnbindAsync();
+                    (IConnectionListener listener, Task acceptLoop) = _transports[i];
+                    tasks[i] = Task.WhenAll(listener.UnbindAsync(cancellationToken).AsTask(), acceptLoop);
```

Comment:
I'm not a fan of having to do Task.WhenAll here between unbinding and the accept loop. Can you clarify why you need to wait on both? How would someone else using these interfaces handle this?

Reply:
To dispose, we need to first unbind so no new connections can enter the accept loop, then we need to await the accept loop so that it drains existing connections before disposing.